### PR TITLE
Rename release profile

### PR DESCRIPTION
**Pull request by cstamas**
As we dumped nx2 staging plugin, rename the profile ID to something more logical: "sisu-release".

### CODE REVIEW

File: org.eclipse.sisu.inject/pom.xml [hunk 1]
```diff
@@ -456,7 +456,7 @@
 
   <profiles>
     <profile>
-      <id>sonatype-oss-release</id>
+      <id>sisu-release</id>
```
Comment (kwin) [inline org.eclipse.sisu.inject/pom.xml:459]: Why not the default profile name `release`?

Comment (cstamas) [reply org.eclipse.sisu.inject/pom.xml:459]: As it may conflict with out of the box profile, and as you see, release plugin explicitly disables it.

Comment (cstamas) [reply org.eclipse.sisu.inject/pom.xml:459]: Also, the release profile used by project was not "default" anyway, this PR does not change in this respect, just shortens its name (and aligns it, with other "release profiles" like `apache-release` or `takari-release` etc).

Comment (kwin) [reply org.eclipse.sisu.inject/pom.xml:459]: I still don't get why we need a project-specific prefix. I like to have the same profile name across all projects, therefore I usually pick `release`. But as it always has been used with some prefix I am fine with keeping it.

**Review by kwin — APPROVED** [review]

**Pull request merged**